File: air_mobile_app/backend/models/TaskModel.js
const mongoose = require('mongoose');

const TaskSchema = new mongoose.Schema({
    title: { type: String, required: true },
    description: String,
    category: String,
    priority: { type: String, enum: ['Low', 'Medium', 'High'], required: true },
    progress: { type: Number, default: 0 },
    date: String,
    time: String,
    image: String,
    status: { type: String, enum: ['Pending', 'In Progress', 'Completed'], default: 'Pending' }
},{ collection: 'Task' });

const Task = mongoose.model('Task', TaskSchema);
module.exports = Task;

File: air_mobile_app/frontend/lib/model/task_model.dart
import 'dart:core';

class TaskModel {
  String? key;
  String? title;
  String? category;
  String? description;
  String? image;
  String? periority;
  String? time;
  String? date;
  String? show;
  int? progress;
  String? status;

  TaskModel(
      {required this.key,
      required this.time,
      required this.date,
      required this.periority,
      required this.description,
      required this.category,
      required this.title,
      required this.image,
      required this.show,
      required this.progress,
      required this.status});

  TaskModel.fromMap(Map<String, dynamic> res) {
    key = res['_id']?.toString();
    title = res['title'];
    category = res['category'];
    description = res['description'];
    image = res['image'];
    periority = res['priority'];
    show = 'yes';
    time = res['time'];
    date = res['date'];
    progress = res['progress'] as int?;
    status = res['status'];
  }

  Map<String, Object?> toMap() {
    return {
      'key': key,
      'title': title,
      'category': category,
      'description': description,
      'image': image,
      'periority': periority,
      'time': time,
      'date': date,
      'show': show,
      'status': status,
      'progress': progress,
    };
  }
}
